aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -254,6 +259,34 @@ module aptos_framework::stake {
         pool_address: address,
     }
 
+    /// Stores transaction fees assigned to validators. All fees are distributed to validators
+    /// at the end of the epoch.
+    struct ValidatorFees has key {
+        fees_table: Table<address, Coin<AptosCoin>>,
+    }
+
+    /// Initializes the resource storing information about collected transaction fees per validator.
+    /// Used by `transaction_fee.move` to initialize fee collection and distribution.
+    public(friend) fun initialize_validator_fees(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+        assert!(
+            !exists<ValidatorFees>(@aptos_framework),
+            error::already_exists(EFEES_TABLE_ALREADY_EXISTS)
+        );
+        move_to(aptos_framework, ValidatorFees { fees_table: table::new() });
+    }
+
+    /// Stores the transaction fee collected to the specified validator address.
+    public(friend) fun add_transaction_fee(validator_addr: address, fee: Coin<AptosCoin>) acquires ValidatorFees {
+        let fees_table = &mut borrow_global_mut<ValidatorFees>(@aptos_framework).fees_table;
```

Comment:
Should we add a safeguard here in case the ValidatorFees resource has not been initialized?

Reply:
Then we would have a coin which we cannot drop (assuming we guard with `if !exists<..>(..) return`? Both `CollectedFeesPerBlock` and `ValidatorFees` are published at the same time, though.

Reply:
Hmm that's true that we can't drop or burn the coins here 🤔 It's fine not to have the safeguard then if those 2 resources are guaranteed to be published together. cc @junkil-park